simulator/host_network_system.go

```diff
@@ -198,9 +200,13 @@ func (s *HostNetworkSystem) UpdateNetworkConfig(req *types.UpdateNetworkConfig)
 func (s *HostNetworkSystem) QueryNetworkHint(req *types.QueryNetworkHint) soap.HasFault {
 	var info []types.PhysicalNicHintInfo
 
-	for _, nic := range s.Host.Config.Network.Pnic {
```

Comment:
This causes the unit test to fail:
```
go test -v -run TestHostNetworkSystem\$ . 
=== RUN   TestHostNetworkSystem
    host_network_system_test.go:174: len=0
--- FAIL: TestHostNetworkSystem (0.01s)
FAIL
FAIL	github.com/vmware/govmomi/simulator	0.298s
FAIL
```

That's because `s.QueryNetworkHintResponse.Returnval` is empty unless vcsim is run with `-load`. Suggest the following change to address that:
```diff
modified   simulator/host_network_system.go
@@ -198,16 +198,14 @@ func (s *HostNetworkSystem) UpdateNetworkConfig(req *types.UpdateNetworkConfig)
 }
 
 func (s *HostNetworkSystem) QueryNetworkHint(req *types.QueryNetworkHint) soap.HasFault {
-	var info []types.PhysicalNicHintInfo
-
-	for _, nic := range s.QueryNetworkHintResponse.Returnval {
-		info = append(info, types.PhysicalNicHintInfo{
-			Device:              nic.Device,
-			Subnet:              nic.Subnet,
-			Network:             nic.Network,
-			ConnectedSwitchPort: nic.ConnectedSwitchPort,
-			LldpInfo:            nic.LldpInfo,
-		})
+	info := s.QueryNetworkHintResponse.Returnval
+
+	if len(info) == 0 {
+		for _, nic := range s.Host.Config.Network.Pnic {
+			info = append(info, types.PhysicalNicHintInfo{
+				Device: nic.Device,
+			})
+		}
 	}
 
 	return &methods.QueryNetworkHintBody{
```

Reply:
@dougm I think we should update UT for this one. The updated code gives correct data for QueryNetworkHintResponse

Reply:
Correct data when vcsim `-load` flag is given, otherwise `s.QueryNetworkHintResponse.Returnval` is nil.

We do similar by returning default data when load/Returnval is not set here:

https://github.com/vmware/govmomi/blob/cf96cab7e22d0f7e3dcfaaf6d079b9491d8af7a9/simulator/environment_browser.go#L62-L68

and here: https://github.com/vmware/govmomi/blob/cf96cab7e22d0f7e3dcfaaf6d079b9491d8af7a9/simulator/dvs.go#L272-L275